src/identity-utils.ts

```diff
@@ -0,0 +1,198 @@
+    xhr: XMLHttpRequest
+): void => {
+    const CACHE_HEADER = 'x-mp-max-age';
+
+    // default the expire timestamp to one day in milliseconds unless a header comes back
+    let expireTimestamp = new Date().getTime() + ONE_DAY_IN_SECONDS * MILLIS_IN_ONE_SEC;
+
+    if (xhr.getAllResponseHeaders().includes(CACHE_HEADER)) {
+        expireTimestamp =
+            parseNumber(xhr.getResponseHeader(CACHE_HEADER)) * MILLIS_IN_ONE_SEC;
+    }
+
+    switch (method) {
+        case Login: 
+        case Identify: 
+            cacheIdentityRequest(
+                method,
+                knownIdentities,
+                expireTimestamp,
+                idCache,
+                xhr 
+            );
+            break;
+        case Modify:
+        case Logout:
+            idCache.purge();
+            break;
+    }
+}
+
+export const cacheIdentityRequest = (
+    method: IdentityAPIMethod,
+    identities: IKnownIdentities,
+    expireTimestamp: number,
+    idCache: BaseVault<Dictionary>,
+    xhr: XMLHttpRequest
+): void => {
+    const cache: Dictionary<ICachedIdentityCall> = idCache.retrieve() || {};
+    const cacheKey = concatenateIdentities(method, identities);
+
+    cache[cacheKey] = { responseText: xhr.responseText, status: xhr.status, expireTimestamp};
+    idCache.store(cache);
+};
+
+// We need to ensure that identities are concatenated in a deterministic way, so
+// we sort the identities based on their enum.
+// we create an array, set the user identity at the index of the user identity type
+export const concatenateIdentities = (
+    method: IdentityAPIMethod,
+    userIdentities: IKnownIdentities
+): string => {
+    const DEVICE_APPLICATION_STAMP = 'device_application_stamp';
+    // set DAS first since it is not an official identity type
+    let cacheKey: string = `${method}:${DEVICE_APPLICATION_STAMP}=${userIdentities.device_application_stamp};`;
+    const idLength: number = Object.keys(userIdentities).length;
+    let concatenatedIdentities: string = '';
+
+    if (idLength) {
+        let userIDArray: Array<string> = new Array();
+        // create an array where each index is equal to the user identity type
+        for (let key in userIdentities) {
+            if (key === DEVICE_APPLICATION_STAMP) {
+                continue;
+            } else {
+                userIDArray[Types.IdentityType.getIdentityType(key)] =
+                    userIdentities[key];
+            }
+        }
+
+        concatenatedIdentities = userIDArray.reduce(
+            (prevValue: string, currentValue: string, index: number) => {
+                const idName: string = Types.IdentityType.getIdentityName(index);
+                return `${prevValue}${idName}=${currentValue};`;
+            },
+            cacheKey
+        );
+    }
+
+    return concatenatedIdentities;
+};
+
+export const hasValidCachedIdentity = (
+    method: IdentityAPIMethod,
+    proposedUserIdentities: IKnownIdentities,
+    idCache?: BaseVault<Dictionary>
```

Comment:
I don't think this should be just a generic `Dictionary` but rather a clearly defined `IdentityCache` Type.